### PR TITLE
Fix filter removal and upload for un-existing filters

### LINKED ISSUE #2060: NoneType on filter get - deleting filter twice
Cater for the case of deleting last filter for an institute / same filter twice?

```
2020-09-01 14:01:23,660 - scout.server.app - ERROR: Exception on /cust002/F0035385/variants [POST] [in /home/hiseq.clinical/miniconda/envs/prod/lib/python3.6/site-packages/flask/app.py:1892]
Traceback (most recent call last):
 File "/home/hiseq.clinical/miniconda/envs/prod/lib/python3.6/site-packages/flask/app.py", line 2447, in wsgi_app
   response = self.full_dispatch_request()
 File "/home/hiseq.clinical/miniconda/envs/prod/lib/python3.6/site-packages/flask/app.py", line 1952, in full_dispatch_request
   rv = self.handle_user_exception(e)
 File "/home/hiseq.clinical/miniconda/envs/prod/lib/python3.6/site-packages/flask/app.py", line 1821, in handle_user_exception
   reraise(exc_type, exc_value, tb)
 File "/home/hiseq.clinical/miniconda/envs/prod/lib/python3.6/site-packages/flask/_compat.py", line 39, in reraise
   raise value
 File "/home/hiseq.clinical/miniconda/envs/prod/lib/python3.6/site-packages/flask/app.py", line 1950, in full_dispatch_request
   rv = self.dispatch_request()
 File "/home/hiseq.clinical/miniconda/envs/prod/lib/python3.6/site-packages/flask/app.py", line 1936, in dispatch_request
   return self.view_functions[rule.endpoint](**req.view_args)
 File "/home/hiseq.clinical/SERVER/apps/scout/git/scout/scout/server/utils.py", line 23, in decorated_function
   context = func(*args, **kwargs)
 File "/home/hiseq.clinical/SERVER/apps/scout/git/scout/scout/server/blueprints/variants/views.py", line 57, in variants
   store, institute_obj, case_obj, user_obj, category, request.form
 File "/home/hiseq.clinical/SERVER/apps/scout/git/scout/scout/server/blueprints/variants/controllers.py", line 621, in populate_filters_form
   filter_obj = store.delete_filter(filter_id, institute_id, current_user.email)
 File "/home/hiseq.clinical/SERVER/apps/scout/git/scout/scout/adapter/mongo/filter.py", line 122, in delete_filter
   user_id, filter_obj.get("display_name"), institute_id
AttributeError: 'NoneType' object has no attribute 'get'
```

### PR DESCRIPTION
**Pull request by northwestwitch**
fix #2060

**How to test**:
1. Install the branch and open the variants page for a case.
1. Create a custom filter
1. Open the same page on another browser tab
1. Delete the filter from one tab and try to delete it from the second tab. It should flash an error message and not crash.
1. Repeat the filter creation and have it displayed on the 2 tabs
1. Remove from the first tab and try to load it from the second tab. Also in this case it should display an error message, no page crash

**Expected outcome**:
The functionality should be working
Take a screenshot and attach or copy/paste the output.

**Review:**
- [x] code approved by DN
- [x] tests executed by DN

### CODE REVIEW

**Comment by northwestwitch** [comment]
Still need to fix a couple of things on this one. Wait for the review @dnil

**Comment by northwestwitch** [comment]
@dnil now it might work as intended!

File: scout/server/blueprints/variants/controllers.py
Comment (dnil) [inline scout/server/blueprints/variants/controllers.py:662]: :+1:

**Review by dnil — APPROVED** [review]
Looks good - thank you!

**Comment by dnil** [comment]
Don't forget CHANGELOG!

**Comment by northwestwitch** [comment]
I think you need to re-approve now that I've updated the changelog. Basically any change after an approving review needs another review. This is a security thing introduced today that seemed a good idea at the moment, but I already start to regret. Shall we remove it?

**Comment by dnil** [comment]
Yes, I think we could largely handle the freedom given by the previous rule. This is going to be a bit cumbersome. We can always revert..

**Review by dnil — APPROVED** [review]

**Comment by dnil** [comment]
I see coveralls was also made mandatory. I have a hunch we will regret that one as well.. 😉

**Pull request merged**
